### PR TITLE
MFTF: Enable Persistent Shopping Cart. Assert Options

### PR DESCRIPTION
**Pull request by DmitryTsymbal**
**Description**
This PR with a test that Enables Persistent Shopping Cart in Admin Panel

**Steps To reproduce:**

1 - Login to admin
2 - Enable Persistent Shopping Cart
3 - Navigate to Stores->Configuration->Customers->Persistent Shopping Cart
4 - Assert additional options

### CODE REVIEW

File: app/code/Magento/Backend/Test/Mftf/Page/AdminConfigurationStoresPage.xml [hunk 1]
```diff
@@ -17,4 +17,5 @@
     <page name="GeneralConfigurationPage" url="admin/system_config/edit/section/general/" area="admin" module="Backend">
         <section name="LocaleOptionsSection"/>
     </page>
+    <page name="PersistentShoppingCartConfig" url="admin/system_config/edit/section/persistent/" module="Customers" area="admin"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Backend/Test/Mftf/Page/AdminConfigurationStoresPage.xml]: 1 Page per file, "Page" suffix is missing.

File: app/code/Magento/Backend/Test/Mftf/ActionGroup/AssertAdminPersistentShoppingCartOptionsAvailableActionGroup.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AssertAdminPersistentShoppingCartOptionsAvailableActionGroup">
+
```
Comment (lbajsarowicz) [inline app/code/Magento/Backend/Test/Mftf/ActionGroup/AssertAdminPersistentShoppingCartOptionsAvailableActionGroup.xml]: Too much responsibility for single action group.
Extract opening page and the section to separate action group.

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Please find the comments

File: app/code/Magento/Backend/Test/Mftf/Test/AdminPersistentShoppingCartEnableTest.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminPersistentShoppingCartEnableTest">
+        <annotations>
+            <features value="Backend"/>
+            <stories value="Enable Persistent Shopping cart"/>
+            <title value="Admin should be able to enable persistent shopping cart"/>
```
Comment (dmytro-ch) [inline app/code/Magento/Backend/Test/Mftf/Test/AdminPersistentShoppingCartEnableTest.xml]:
```suggestion
            <title value="Admin should be able to manage persistent shopping cart settings"/>
```

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
Hello @DmitryTsymbal, could you please rename the test to "AdminPersistentShoppingCartSettingsTest" or something similar, just to make sure the name is a bit more relevant?
Thank you!

**Comment by DmitryTsymbal** [comment]
Thank you @dmytro-ch and @lbajsarowicz for review.

I have added requested changes

File: app/code/Magento/Backend/Test/Mftf/ActionGroup/AssertAdminPersistentShoppingCartOptionsAvailableActionGroup.xml [hunk 4]
```diff
@@ -9,8 +9,6 @@
 <actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
               xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
     <actionGroup name="AssertAdminPersistentShoppingCartOptionsAvailableActionGroup">
-
-        <amOnPage url="{{PersistentShoppingCartConfig.url}}" stepKey="navigateToPersistencePage"/>
         <conditionalClick selector="{{AdminPersistentShoppingCartSection.DefaultLayoutsTab}}" dependentSelector="{{AdminPersistentShoppingCartSection.CheckIfTabExpand}}" visible="true" stepKey="clickTab"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Backend/Test/Mftf/ActionGroup/AssertAdminPersistentShoppingCartOptionsAvailableActionGroup.xml]: You can also move that line to AdminNavigateToPersistentShoppingCartSettingsActionGroup, as a result of that action group we have configuration grid ready to work

File: app/code/Magento/Backend/Test/Mftf/Page/AdminConfigurationPersistentShoppingCartPage.xml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<pages xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Page/etc/PageObject.xsd">
+    <page name="AdminConfigurationPersistentShoppingCartPage" url="admin/system_config/edit/section/persistent/" module="Customers" area="admin"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Backend/Test/Mftf/Page/AdminConfigurationPersistentShoppingCartPage.xml]: Add section there AdminPersistentShoppingCartSection

File: app/code/Magento/Backend/Test/Mftf/Test/AdminPersistentShoppingCartSettingsTest.xml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+            <stories value="Enable Persistent Shopping cart"/>
+            <title value="Admin should be able to manage persistent shopping cart settings"/>
+            <description value="Admin should be able to enable persistent shopping cart in Magento Admin backend and see additional options"/>
+            <group value="backend"/>
+        </annotations>
+
+        <before>
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+            <magentoCLI stepKey="enablePersistentShoppingCart" command="config:set persistent/options/enabled 1"/>
+            <magentoCLI stepKey="cacheFlush" command="cache:flush"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Backend/Test/Mftf/Test/AdminPersistentShoppingCartSettingsTest.xml]: cache:clean config is enough

File: app/code/Magento/Backend/Test/Mftf/Test/AdminPersistentShoppingCartSettingsTest.xml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        </before>
+        <after>
+            <magentoCLI stepKey="disablePersistentShoppingCart" command="config:set persistent/options/enabled 0"/>
+            <magentoCLI stepKey="cacheFlush" command="cache:flush"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Backend/Test/Mftf/Test/AdminPersistentShoppingCartSettingsTest.xml]: Same

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Some more changes

File: app/code/Magento/Backend/Test/Mftf/ActionGroup/AssertAdminPersistentShoppingCartOptionsAvailableActionGroup.xml (hunk 4 above)
Comment (DmitryTsymbal) [reply app/code/Magento/Backend/Test/Mftf/ActionGroup/AssertAdminPersistentShoppingCartOptionsAvailableActionGroup.xml]: Done

File: app/code/Magento/Backend/Test/Mftf/Page/AdminConfigurationPersistentShoppingCartPage.xml (hunk 5 above)
Comment (DmitryTsymbal) [reply app/code/Magento/Backend/Test/Mftf/Page/AdminConfigurationPersistentShoppingCartPage.xml]: Done

File: app/code/Magento/Backend/Test/Mftf/Test/AdminPersistentShoppingCartSettingsTest.xml (hunk 6 above)
Comment (DmitryTsymbal) [reply app/code/Magento/Backend/Test/Mftf/Test/AdminPersistentShoppingCartSettingsTest.xml]: Done

File: app/code/Magento/Backend/Test/Mftf/Test/AdminPersistentShoppingCartSettingsTest.xml (hunk 7 above)
Comment (DmitryTsymbal) [reply app/code/Magento/Backend/Test/Mftf/Test/AdminPersistentShoppingCartSettingsTest.xml]: Done

File: app/code/Magento/Backend/Test/Mftf/Test/AdminPersistentShoppingCartSettingsTest.xml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+            <stories value="Enable Persistent Shopping cart"/>
+            <title value="Admin should be able to manage persistent shopping cart settings"/>
+            <description value="Admin should be able to enable persistent shopping cart in Magento Admin backend and see additional options"/>
+            <group value="backend"/>
+        </annotations>
+
+        <before>
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+            <magentoCLI stepKey="enablePersistentShoppingCart" command="config:set persistent/options/enabled 1"/>
+            <magentoCLI stepKey="cacheClean" command="cache:clean"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Backend/Test/Mftf/Test/AdminPersistentShoppingCartSettingsTest.xml]: You can specify to clean only `config` cache.

```suggestion
            <magentoCLI stepKey="cacheClean" command="cache:clean config"/>
```

File: app/code/Magento/Backend/Test/Mftf/Test/AdminPersistentShoppingCartSettingsTest.xml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+        </before>
+        <after>
+            <magentoCLI stepKey="disablePersistentShoppingCart" command="config:set persistent/options/enabled 0"/>
+            <magentoCLI stepKey="cacheClean" command="cache:clean"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Backend/Test/Mftf/Test/AdminPersistentShoppingCartSettingsTest.xml]:
```suggestion
            <magentoCLI stepKey="cacheClean" command="cache:clean config"/>
```

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Cleaning / flushing caches cause test performance overhead.
That's why I focus a lot on it :-) Please take a look at the suggestions.

**Comment by dmytro-ch** [comment]
> Cleaning / flushing caches cause test performance overhead.
> That's why I focus a lot on it :-) Please take a look at the suggestions.

@lbajsarowicz good notice. I guess we don't even need to refresh the `config` cache, considering we don't actually interact with the frontend.

**Comment by lbajsarowicz** [comment]
I assume that you change the value just to have the options expanded.
Which is fine for the test purpose.

Now you get green :-)

**Review by lbajsarowicz — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7088](https://jira.corp.magento.com/browse/ENGCOM-7088) has been created to process this Pull Request

**Review by dmytro-ch — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-7088](https://jira.corp.magento.com/browse/ENGCOM-7088) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Comment by engcom-Charlie** [comment]
Failed functional tests not related to the changes in this PR.

**Pull request merged**